Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp

```diff
@@ -60,259 +56,260 @@ BluetoothRealDevice::BluetoothRealDevice(EmulationKernel& ios, const std::string
+        DEBUG_LOG_FMT(IOS_WIIMOTE, "HCI_CMD_SNIFF_MODE");
       }
-      NOTICE_LOG_FMT(IOS_WIIMOTE, "Using device {:04x}:{:04x} (rev {:x}) for Bluetooth: {} {} {}",
-                     device_descriptor.idVendor, device_descriptor.idProduct,
-                     device_descriptor.bcdDevice, reinterpret_cast<char*>(manufacturer),
-                     reinterpret_cast<char*>(product), reinterpret_cast<char*>(serial_number));
-      m_is_wii_bt_module =
-          device_descriptor.idVendor == 0x57e && device_descriptor.idProduct == 0x305;
-      return false;
+
+      const auto payload = memory.GetSpanForAddress(cmd->data_address).first(cmd->length);
+      m_lib_usb_bt_adapter->ScheduleControlTransfer(cmd->request_type, cmd->request, cmd->value,
+                                                    cmd->index, payload, GetTargetTime());
     }
-    return true;
-  });
-  if (ret != LIBUSB_SUCCESS)
-  {
-    m_last_open_error =
-        Common::FmtFormatT("GetDeviceList failed: {0}", LibusbUtils::ErrorWrap(ret));
+    return IPCReply{cmd->length};
   }
-
-  if (m_handle == nullptr)
+  // ACL data (incoming or outgoing)
+  case USB::IOCTLV_USBV0_BLKMSG:
   {
-    if (m_last_open_error.empty())
+    auto cmd = std::make_unique<USB::V0BulkMessage>(GetEmulationKernel(), request);
+    if (cmd->endpoint == ACL_DATA_IN)
     {
-      CriticalAlertFmtT(
-          "Could not find any usable Bluetooth USB adapter for Bluetooth Passthrough.\n\n"
-          "The emulated console will now stop.");
+      m_acl_endpoint = std::move(cmd);
+      TryToFillACLEndpoint();
+      return std::nullopt;
     }
-    else
+
+    auto& memory = GetSystem().GetMemory();
+    const auto payload = memory.GetSpanForAddress(cmd->data_address).first(cmd->length);
+    m_lib_usb_bt_adapter->ScheduleBulkTransfer(cmd->endpoint, payload, GetTargetTime());
+    break;
+  }
+  // Incoming HCI events
+  case USB::IOCTLV_USBV0_INTRMSG:
+  {
+    auto cmd = std::make_unique<USB::V0IntrMessage>(GetEmulationKernel(), request);
+    if (cmd->endpoint == HCI_EVENT)
     {
-      CriticalAlertFmtT(
-          "Could not find any usable Bluetooth USB adapter for Bluetooth Passthrough.\n"
-          "The following error occurred when Dolphin tried to use an adapter:\n{0}\n\n"
-          "The emulated console will now stop.",
-          m_last_open_error);
+      m_hci_endpoint = std::move(cmd);
+      TryToFillHCIEndpoint();
+      return std::nullopt;
     }
-    Core::QueueHostJob(&Core::Stop);
-    return IPCReply(IPC_ENOENT);
+
+    ERROR_LOG_FMT(IOS_WIIMOTE, "IOCTLV_USBV0_INTRMSG: Unknown endpoint: 0x{:02x}", cmd->endpoint);
+  }
+  default:
+    ERROR_LOG_FMT(IOS_WIIMOTE, "IOCtlV: Unknown request: 0x{:08x}", request.request);
+    break;
   }
 
-  return Device::Open(request);
+  return IPCReply{IPC_SUCCESS};
 }
 
-std::optional<IPCReply> BluetoothRealDevice::Close(u32 fd)
+void BluetoothRealDevice::Update()
+{
+  if (m_hci_endpoint != nullptr)
```

Comment:
Do we ever expect `Update()` will increase in complexity?  This is a huge nitpick but these checks feel like they'd be better in the function where they are being referenced in.  Ex:

```
void BluetoothRealDevice::TryToFillHCIEndpoint()
{
  if (!m_hci_endpoint)
    return;

  ...
}
```

Reply:
Yeah, I suppose it's more organized like that.